libsql/src/replication/remote_client.rs

```diff
@@ -107,8 +172,12 @@ impl RemoteClient {
         Ok(new_session)
     }
 
-    async fn do_handshake_with_prefetch(&mut self) -> (Result<bool, Error>, Duration) {
+    async fn do_handshake_with_prefetch(&mut self) -> (Result<(), Error>, Duration) {
         tracing::info!("Attempting to perform handshake with primary.");
+        if let Some((Ok(frames), _)) = &self.prefetched_batch_log_entries {
+            let bytes = frames.get_ref().frames.iter().map(|f| f.data.len() as u64).sum();
```

Comment:
do we even need to sum the length, can we just assume that each frame is 4kb and just take frames.len() * 4kb?

Reply:
I'm not sure but it is a good question.

Reply:
Added a TODO to check if it's correct.